botocore/docs/sharedexample.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import re
+
+
+class SharedExampleBuilder(object):
+    def __init__(self, params, operation_name, comments=None, is_input=True):
+        self._params = params
+        self._operation_name = operation_name
+        self._comments = comments
+        self._is_input = is_input
+
+    def build_example_code(self, prefix=''):
+        if self._is_input:
+            return prefix + self._visit(self._params, '', [], is_param=True)
+        else:
+            return self._visit(self._params, '', [], is_param=False)
+
+    def _visit(self, value, indent, path, is_param=False):
+        if is_param:
+            return self._visit_param(value, indent, path)
+        if isinstance(value, dict):
+            return self._visit_dict(value, indent, path)
+        if isinstance(value, list):
+            return self._visit_list(value, indent, path)
+
+        try:
+            if isinstance(value, basestring):
```

Comment:
Oh totally missed catching NameError.  The preferred way to handle py2/py3 differences is to pull from compat.py.  You should be able to use `compat.six.string_types` which works for py2/py3.